ecosystem/node-checker/src/evaluators/direct/api/transaction_availability.rs

```diff
@@ -104,88 +75,127 @@ impl Evaluator for TransactionAvailabilityEvaluator {
     /// baseline produced after a delay. We confirm that the transactions are
     /// same by looking at the version.
     async fn evaluate(&self, input: &Self::Input) -> Result<Vec<EvaluationResult>, Self::Error> {
+        let oldest_baseline_version = input
+            .baseline_index_response
+            .ledger_info
+            .oldest_ledger_version
+            .0;
+        let oldest_target_version = input
+            .target_index_response
+            .ledger_info
+            .oldest_ledger_version
+            .0;
+        let latest_baseline_version = input.baseline_index_response.ledger_info.ledger_version.0;
+        let latest_target_version = input.target_index_response.ledger_info.ledger_version.0;
+
+        // Get the oldest ledger version between the two nodes.
+        let oldest_shared_version = max(oldest_baseline_version, oldest_target_version);
+
+        // Get the least up to date latest ledger version between the two nodes.
+        let latest_shared_version = min(latest_baseline_version, latest_target_version);
+
+        // Ensure that there is a window between the oldest shared version and
+        // latest shared version. If there is not, it will not be possible to
+        // pull a transaction that both nodes have.
+        if oldest_shared_version > latest_shared_version {
+            return Ok(vec![self.build_evaluation_result(
+                "Unable to pull transaction from both nodes".to_string(),
+                0,
+                format!(
+                    "We were unable to find a ledger version window between \
+                        the baseline and target nodes. The oldest and latest \
+                        ledger versions on the baseline node are {} and {}. \
+                        The oldest and latest ledger versions on the target \
```

Comment:
nit: I would cut this out completely:
```
Taking the max of the oldest ledger versions ({}) and the min of the newest ledger versions ({}) means there is no ledger version window in which there is a transaction available from the API of both nodes.
```

Seems unnecessary for your average user.